classes/form/management_displaynames_flavors_form.php

```diff
@@ -0,0 +1,133 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tiny_elements\form;
+
+use core_form\dynamic_form;
+use tiny_elements\local\utils;
+use context;
+
+/**
+ * Form for bulk editing displaynames of flavors
+ *
+ * @package    tiny_elements
+ * @copyright  2025 ISB Bayern
+ * @author     Stefan Hanauska <stefan.hanauska@csg-in.de>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class management_displaynames_flavors_form extends dynamic_form {
```

Comment:
why do these forms not inherit from base_form?

Reply:
Because it's necessary to extend dynamic_form if you want to use the form in a modal.